src/app/GitCommands/Git/GitModule.cs

```diff
@@ -3191,18 +3185,35 @@ public IReadOnlyList<string> GetFullTree(string id)
                 .Split(Delimiters.NullAndLineFeed);
         }
 
-        public IEnumerable<INamedGitItem> GetTree(ObjectId? commitId, bool full)
+        public IEnumerable<INamedGitItem> GetTree(ObjectId? commitId, bool full, CancellationToken cancellationToken = default)
+            => GetGitItemTree(commitId, full, cancellationToken);
+
+        public IEnumerable<GitItem> GetGitItemTree(ObjectId? commitId, bool full, CancellationToken cancellationToken = default)
         {
-            GitArgumentBuilder args = new("ls-tree")
+            bool isArtificial = commitId?.IsArtificial is true;
+            if (isArtificial && !full)
             {
-                "-z",
-                { full, "-r" },
-                commitId
-            };
+                throw new ArgumentOutOfRangeException(nameof(full), "Artificial commit requires fu.");
```

Comment:
There is no need to leave a nit for me. :)
```suggestion
                throw new ArgumentOutOfRangeException(nameof(full), "Artificial commit requires 'full'.");
```